hfs/xattr.c

```diff
@@ -251,6 +268,7 @@ size_t getAttribute(Volume* volume, uint32_t fileID, const char* name, uint8_t**
 			return size;
 		default:
 			fprintf(stderr, "unsupported attribute node format\n");
+			free(record);
```

Comment:
I feel like the code above would be more readable if it were factored out, like:
```
record = (HFSPlusAttrRecord*) search(volume->attrTree, (BTKey*)(&key), &exact, NULL, NULL);
size = MaybeFillAttributeData(exact, record, data);

if (record) {
    free(record);
}

return size;
```
This is just a nit though, the fix itself is sound and I understand you might not want to alter this code too much.

Reply:
Yes, this is fine, thank you!